Update paper.md (#141)
OceanBioME/OceanBioME.jl: paper/paper.md

To date, about 25% of anthropogenic carbon emissions have been taken up by the ocean [@Friedlingstein2022]. 
This occurs through complex interactions between physics, chemistry, and biology, much of which is poorly understood. 
Due to the vast size of the ocean and the sparsity of data; modelling and data assimilation play a vital role in quantifying the ocean carbon cycle. 
Traditionally ocean biogeochemical (BGC) modelling involves large and inflexible code bases written in high-performance but low-level languages. 

> Using and modifying these models usually requires knowledge of the details of the numerical implementation (e.g. the grid layout and time-stepping schemes). 

Reviewer: ```suggestion
Both using and also modifying these models usually requires knowledge of the details of the numerical implementation (e.g. the grid layout and time-stepping schemes). 
```

reads better? not sure... feel free to ignore

Author: I kind of like the original here

Reviewer: okie!

This necessitates adaptable, easy-to-use, and verifiable BGC modelling tools which can be used to assess OCDR strategies at the fast pace with which they are being developed [@NationalAcademies2022].
We have built ``OceanBioME.jl`` to meet these challenges by creating a tool that provides a modular interface to the different components, within the ocean modelling framework provided by ``Oceananigans.jl``.
This allows easy access to a suite of biogeochemical models ranging from simple idealized to full-complexity models.
``Oceananignas.jl`` and ``OceanBioME.jl`` are built from the ground-up to exploit the power of graphical processor units (GPUs), while also retaining the ability to run on CPUs.
The flexibility of the ``Oceananigans.jl`` framework allows ``OceanBioME.jl`` to be applied across a wide range of scales and use cases, including small-scale large-eddy simulations and regional and global models.

# Summary

``OceanBioME.jl`` is a flexible modelling environment written in Julia [@julia] for simulating the coupled interactions between ocean biogeochemistry, carbonate chemistry, and physics.

> ``OceanBioME.jl`` can be used as a stand-alone box model, or integrated into ``Oceananigans.jl`` [@Oceananigans] for coupled physical/biogeochemical simulations in one, two, or three dimensions.

Reviewer: ```suggestion
``OceanBioME.jl`` can be used as a stand-alone box model, or integrated into ``Oceananigans.jl`` [@Oceananigans] for coupled physical-biogeochemical simulations in one, two, or three dimensions.
```

As a result, ``OceanBioME.jl`` and ``Oceananigans.jl`` can be used to simulate the biogeochemical response across an enormous range of scales: from surface boundary layer turbulence at the sub-meter scale to eddying global ocean simulations at the planetary scale, and on computational systems ranging from laptops to supercomputers.

> An example of a problem involving small-scale flow features is shown in \autoref{eady}, which shows a simulation of a sub-mesoscale eddy in a 1km x 1km horizontal domain with an intermediate complexity biogeochemical model and a kelp growth model solved along the trajectories of drifting buoys (a list of of examples shown in this paper and links to source code are given at the end of the paper).

Reviewer: > "is shown in \autoref{eady}, which shows"

"show" appears twice too soon...?

```suggestion
An example of a problem involving small-scale flow features is showcased in \autoref{eady}, which shows a simulation of a sub-mesoscale eddy in a 1km x 1km horizontal domain with an intermediate complexity biogeochemical model and a kelp growth model solved along the trajectories of drifting buoys (a list of of examples shown in this paper and links to source code are given at the end of the paper).
```

The increase in organic carbon concentration in the centre of the eddy can be seen, as well as carbon being subducted (most visible in the xz face in the organic carbon).
Points on the surface represent the kelp particle positions, with the colour representing the range of frond size.
Panel (b) shows the carbon stored in each kelp frond, highlighting the variability depending on the nutrition availability in each particle's location history.
Figure made with ``Makie.jl`` [@makie]. \label{eady}](eady_example.png)

``OceanBioME.jl`` is built with a highly modular design that allows user control and customization.
There are three distinct module types implemented in ``OceanBioME.jl``:

- First, we provide tracer-based ecosystem modules in `AdvectedPopulations` as a set of coupled ordinary differential equations (ODEs) which evolve the concentration of the tracer.
These equations can be solved by ``OceanBioME.jl`` as box models.
This is useful for both testing and running simple educational models and for understanding the behaviour of biogeochemical models without the effects of the physical dynamics.

> The same tracer equations can then be seamlessly integrated into ``Oceananigans.jl`` to add the effects of advection and diffusion.

Reviewer: ```suggestion
The same tracer equations can then be seamlessly integrated into ``Oceananigans.jl`` to further include the effects of advection and diffusion.
```

not sure if this is better... feel free to ignore

Using and modifying these models usually requires knowledge of the details of the numerical implementation (e.g. the grid layout and time-stepping schemes). 
Most traditional BGC models also require CPU-based supercomputers to run quickly, which presents a barrier to experimentation and innovation.

One area where novel ideas must be explored with BGC codes is assessing ocean carbon dioxide removal (OCDR) strategies. Quantifying the effectiveness and identifying the impacts of OCDR is challenging due to the aforementioned complexity of the ocean BGC system.
Moreover, field trials of OCDR interventions are generally small-scale and targeted, while the intervention required to have a climate-scale impact is regional or global.
This necessitates adaptable, easy-to-use, and verifiable BGC modelling tools which can be used to assess OCDR strategies at the fast pace with which they are being developed [@NationalAcademies2022].
We have built ``OceanBioME.jl`` to meet these challenges by creating a tool that provides a modular interface to the different components, within the ocean modelling framework provided by ``Oceananigans.jl``.

> Previously implmenting biogeochemical models in ``Oceananigans.jl`` required the user to add forcing terms and boundary conditions to generic tracers [@Simoes-Sousa2022].

Reviewer 2: ```suggestion
Previously implementing biogeochemical models in ``Oceananigans.jl`` required the user to add forcing terms and boundary conditions to generic tracers [@Simoes-Sousa2022].
```

Author: good catch, thanks @iuryt

A key metric for the validity of biogeochemical systems is the conservation of elements such as carbon and nitrogen in the system.
We therefore continuously test the implemented models in a variety of simple scenarios (i.e. isolated, with/without air-sea flux, with/without sediment) to ensure that conservation conditions are met, and we will continue to add tests for any models.
Additionally, we check ``OceanBioME.jl`` utilities through standard tests such as comparison to analytical solutions for light attenuation, and conservation of tracers for active particle exudation and sinking.
We plan to expand the features of ``OceanBioME.jl`` in the future and very much welcome user contributions.

Finally, this software is currently facilitating multiple research projects into ocean CDR which would have been significantly harder with other models.
For example, Chen et al. (in prep.) uses the active particle coupling provided to investigate the effects of location and planting density of kelp in the open ocean on their carbon drawdown effect, as in the example above.

> Strong-Wright et al. (n prep.) uses the coupling of both the biogeochemistry and easy interface to couple the physics to study flow interactions with a fully resolved giant kelp forest model including the effects on nutrient transport and distribution.

Reviewer: is there anything we could cite here? No worries if not... But even a GitHub repo might be helpful because then the GitHub issue will include the proper citation of these papers when they are out and people would be able to find them in the future.

Reviewer 3: My code for that paper is in a git repo but its not tidy or public yet, perhaps I should just make a place holder one that we can link to and then fill it when its ready?

Reviewer: Or do nothing also and leave as is -- it was just a suggestion.

Reviewer: We can definitely merge this PR now with "in prep." and if you decide to do something add the ref?

Author: good catch, thanks.  Will merge now
